src/cluster.c

```diff
@@ -2765,13 +2757,21 @@ valkeyClusterAsyncContext *valkeyClusterAsyncContextInit(const valkeyClusterOpti
         return NULL;
     }
 
-    acc = vk_calloc(1, sizeof(valkeyClusterAsyncContext));
+    acc = vk_realloc(cc, sizeof(valkeyClusterAsyncContext));
```

Comment:
We could allocate once. It may be even simpler.

Before we get here, we do `cc = valkeyClusterContextInit(options);`. `valkeyClusterContextInit` is internal so we can change it. We can remove the allocation from it and change it to a void function and we can pass the allocation to it, like this:

```
    acc = vk_calloc(1, sizeof(valkeyClusterAsyncContext));
    valkeyClusterContextInit((valkeyClusterContext *)acc, options);
    if (acc->cc.err != 0) {
        ...
    }
```

WDYT?

Reply:
Thanks, I believe it got better changing this.